cristin-project/src/main/java/no/unit/nva/cristin/projects/model/cristin/adapter/CristinProjectCreatorToNvaContributor.java

```diff
@@ -0,0 +1,52 @@
+package no.unit.nva.cristin.projects.model.cristin.adapter;
+
+import static java.util.Collections.emptyList;
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+import java.util.Optional;
+import java.util.function.Function;
+import no.unit.nva.cristin.model.CristinPerson;
+import no.unit.nva.cristin.projects.model.nva.NvaContributor;
+import no.unit.nva.cristin.projects.model.nva.Person;
+
+public class CristinProjectCreatorToNvaContributor extends CristinPersonToNvaContributorCommon
+    implements Function<CristinPerson, NvaContributor> {
+
+    public static final String PRO_CREATOR = "PRO_CREATOR";
+
+    @Override
+    public NvaContributor apply(CristinPerson cristinPerson) {
+        return extractCreator(cristinPerson);
+    }
+
+    private NvaContributor extractCreator(CristinPerson creator) {
+        return Optional.ofNullable(creator)
+                   .stream()
+                   .peek(this::addRoleDataIfMissing)
+                   .flatMap(this::generateRoleBasedContribution)
+                   .findAny()
+                   .or(() -> creatorWithoutAffiliation(creator))
+                   .orElse(null);
+    }
+
+    private void addRoleDataIfMissing(CristinPerson cristinPerson) {
```

Comment:
Can't this be done using `map` in the stream if you have, e.g.

```
    private CristinPerson addRoleDataIfMissing(CristinPerson cristinPerson) {
      return isNull(cristinPerson.getRoles()) 
          ? cristinPerson.setRoles(emptyList()) 
          : cristinPerson.getRoles().forEach(role -> role.setRoleCode(PRO_CREATOR));
  }
``` ?

Reply:
Will check. Using java 17

Reply:
What do you think? Is it better to use adapter classes to keep the model clean. in case we change backend later?